spark/src/main/scala/geotrellis/spark/summary/polygonal/RDDPolygonalSummary.scala

```diff
@@ -0,0 +1,143 @@
+
+import scala.reflect.ClassTag
+
+object RDDPolygonalSummary {
+  /**
+    * Perform a polygonal summary over all geometries in geometryRdd using data from rasterRdd
+    *
+    * Helper methods are provided in [[RDDPolygonalSummaryMethods]] which handle some common
+    * derived cases, such as only performing a summary for a single geometry across rasterRdd.
+    *
+    * The provided visitor is cloned for each individual polygonal summary operation on each
+    * geometry in the input. Do not rely on private state within the passed visitor to persist
+    * to each summary operation.
+    *
+    * @param rasterRdd
+    * @param geometryRdd
+    * @param visitor
+    * @param options Rasterizer.Options for this polygonal summary operation
+    * @param semigroupR
+    * @param tagR
+    * @tparam R Type of the summary being performed
+    * @tparam T Type of the Raster contained in rasterRdd
+    * @return An RDD containing each unique input geometry and it's summary result R
+    */
+  def apply[R, T <: CellGrid[Int]](
+      rasterRdd: RDD[(SpatialKey, T)] with Metadata[TileLayerMetadata[SpatialKey]],
+      geometryRdd: RDD[Geometry],
+      visitor: GridVisitor[Raster[T], R],
+      options: Rasterizer.Options
+  )(implicit semigroupR: Semigroup[R],
+    tagR: ClassTag[R]): RDD[Feature[Geometry, PolygonalSummaryResult[R]]] = {
+    val layout = rasterRdd.metadata.layout
+
+    val keyedGeometryRdd: RDD[(SpatialKey, Feature[Geometry, UUID])] =
+      geometryRdd.flatMap { geom =>
+        val uuid = UUID.randomUUID
+        val keys: Set[SpatialKey] = layout.mapTransform.keysForGeometry(geom)
+        keys.map(k => (k, Feature(geom, uuid)))
+      }
+
+    // Combine geometries and rasters by SpatialKey, then loop partitions
+    //  and perform the polygonal summary on each combination
+    val joinedRdd = keyedGeometryRdd.join(rasterRdd)
+    val featuresWithSummaries: RDD[(UUID, Feature[Geometry, PolygonalSummaryResult[R]])] =
+      joinedRdd.map { data => {
+        val spatialKey = data._1
+        val feature: Feature[Geometry, UUID] = data._2._1
+        val tile: T = data._2._2
+        val extent: Extent = layout.mapTransform.keyToExtent(spatialKey)
+        val raster: Raster[T] = Raster(tile, extent)
+        val result: PolygonalSummaryResult[R] =
+          raster.polygonalSummary(feature.geom, visitor.getClass.newInstance, options)
+        (feature.data, Feature(feature.geom, result))
+      }
+    }
+
+    // Geometries may have been split across SpatialKey leading to multiple UUID keys.
+    // Combine by UUID and then drop the UUID, which was added within this method
+    featuresWithSummaries
+      .reduceByKey { (a, b) => Feature(a.geom, a.data.combine(b.data)) }
+      .map { _._2 }
+  }
+
+  trait RDDPolygonalSummaryMethods[R, T <: CellGrid[Int]]
+      extends MethodExtensions[RDD[(SpatialKey, T)] with Metadata[TileLayerMetadata[SpatialKey]]] {
+    def polygonalSummary[R](geometryRdd: RDD[Geometry],
```

Comment:
These signatures look good. They need some minimal scaladoc, possibly with `@see RDDPolygonalSummary`. These are going to be the things that show up in tools like IDEA and Metals and we can do a better job providing some hints there.